addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/CsrfTokenScanRuleUnitTest.java

```diff
@@ -98,6 +99,27 @@ public void shouldNotProcessWithoutForm() throws Exception {
         assertThat(httpMessagesSent, hasSize(0));
     }
 
+    @Test
+    public void shouldNotProcessJavaScript() throws Exception {
```

Comment:
`shouldNotProcessNonHtml` would be more accurate, it's not just JavaScript. It could also be parameterized with more types.